### PR TITLE
Adding Babi dataset - English version

### PR DESCRIPTION
**Pull request by thomwolf**
Adding the English version of bAbI.

Samples are taken from ParlAI for consistency with the main users at the moment.

### CODE REVIEW

File: datasets/babi/README.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+---
+YAML tags:
```
Comment (lhoestq) [inline datasets/babi/README.md:2]:
```suggestion
```

File: datasets/babi/babi.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, f"qa{task}_valid.txt"),
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath):
+        """ Yields examples. """
+        pattern = re.compile(r"(\d+)\s([^\t]+)\t?(.*)?")
+        with open(filepath) as f:
```
Comment (lhoestq) [inline datasets/babi/babi.py:155]:
```suggestion
        with open(filepath, encoding="utf-8") as f:
```

**Review by lhoestq — COMMENTED** [review]
Cool thanks !

It would be nice to make the dummy files lighter though.
It should be possible by removing some unused data files. For example for the dummy data zip file of `taskk_10_with_10k_examples` we can remove the data files used for `task_11_with_1k_examples`.

**Comment by thomwolf** [comment]
Replaced by #1126

**Pull request closed without merging**